Feat: Add Dark Mode Support #189

trivedikavya opened the pull request:
**Description**
This PR addresses issue #189 by implementing a fully functional Dark Mode for the GSoC Organizations site. It introduces a global theme management system using Redux, CSS variables for dynamic styling, and specific overrides for Semantic UI components to ensure a seamless visual experience in both Light and Dark modes.

**Key Features**
Global Theme Toggle: Added a button in the Desktop header (next to the search bar) and Mobile toolbar to switch between Light and Dark modes.

**Persistent Preference**: The user's theme choice is saved in localStorage, so it remembers the setting on reload.

**Smart Styling**:

Used CSS Variables (--bg-color, --text-color, --card-bg, etc.) for efficient theming.

Implemented Neumorphic Shadows for organization and project cards in Dark Mode to add depth without "white glow" artifacts.

**Smooth Transitions**: Added CSS transitions to backgrounds and borders for a soft switching effect.

**Component Overrides**:

**Sidebar**: Fully adapts to dark mode with updated text and border colors.

**Search Bar**: Customized to blend perfectly with the dark header.

**Modals & Filters**: Fixed "View All" popups and filter checklists to be readable in dark mode.

**Mobile Experience**: The mobile toolbar gets a distinct translucent dark background (rgba(0, 0, 0, 0.95)).

**Files Changed**
Core Logic & State
_src/store/theme.js_: Created Redux slice for managing theme state.

_src/store/index.js_: Registered the theme reducer.

_src/components/theme-provider.jsx_: Created a wrapper to apply the data-theme attribute to the body.

_wrap-with-provider.js_: Wrapped the root element with ThemeProvider.

Components & UI
_src/components/theme-toggle.jsx_: Created the toggle button component with adaptive styles (Sun/Moon icons).

_src/layouts/desktop/layout.jsx_: Integrated the toggle button and converted the search container to Flexbox for proper alignment.

_src/layouts/mobile/layout.jsx_: Added the toggle button to the mobile toolbar.

Styling (CSS)
_src/layouts/layout.css_: Defined global CSS variables and added overrides for Semantic UI components (Cards, Inputs, Modals).

_src/layouts/desktop/layout.css_: Fixed search bar spacing and colors.

_src/components/sidebar.css_: Updated sidebar background and text colors.

_src/components/org-card.css_ & _project-card.css_: Added hover effects and shadow variables.

_src/components/org-info.css, projects-graph.css, organization.css_: Added specific dark mode borders and shadows.

_src/components/mobile/toolbar.css_: Custom dark background for mobile.

**Screenshots**
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/532aef80-277c-444a-8fde-3611100ce49c" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/f9792d35-d0ea-41fb-9057-984ef928fbbd" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/b2a1345a-1873-4a01-9ab6-75c65ebec1c7" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/0e87fbe5-b7a6-4711-a820-b514bf05261d" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/32298b64-8baa-4334-abeb-73ed4da572bd" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/08a72aab-965a-47ab-944f-55ded54cecfc" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/8d7a9eda-24d7-4438-972a-ca69abba255a" />
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/c80c5327-201e-4428-a363-2a42461e1312" />

closes #189

trivedikavya committed:
Trying

trivedikavya committed:
Working Bro Many Things are Left Still Working

trivedikavya committed:
Main page Perfactly Done

trivedikavya committed:
Icon setup Done

trivedikavya committed:
All done I think so

nishantwrp reviewed (commented):
Hi @trivedikavya, thanks for the dark mode support. Looks neat. But I think we need to think of some solution for how to transform the images to match the dark mode AI before submitting this.

Can you come up with some ideas? Happy to chat with you over a call as well if interested

trivedikavya committed:
Option 1: Smart Invert

trivedikavya commented:
@nishantwrp Done Sir

trivedikavya committed:
Sidebar Theming, Org Card Hover, and Organization Page Dark Mode UI

trivedikavya commented:
hello @nishantwrp sir

1. Fixed Sidebar Theming (Light Mode)
Restored the original sidebar background color (#f5f7f9) and border style in Light Mode to match the original site design.
Ensured the sidebar correctly switches to the dark theme variables in Dark Mode.

2. Fixed Organization Card Hover (Dark Mode)
Resolved an issue where hovering over Organization Cards in Dark Mode showed no visual feedback.
Added a specific "glow" shadow effect for Dark Mode hover states to make interactions visible.

3. Fixed Organization Details Page (Dark Mode)
The Organization Details pages (e.g., /organization/freecad/) previously had hardcoded white backgrounds and text colors, breaking the UI when Dark Mode was enabled.
Updated organization.css, org-info.css, and projects-graph.css to use theme variables, ensuring these pages now fully support Dark Mode.

Screenshots & Verification:

1. Sidebar Restoration (Light Mode)

Original (Reference):
<img width="1920" height="972" alt="Screenshot (76)" src="https://github.com/user-attachments/assets/9a28b836-eef6-4302-af70-8aed687a0c25" />
My Fix:
<img width="1920" height="972" alt="Screenshot (77)" src="https://github.com/user-attachments/assets/f99fe844-f622-4b60-9443-c02b4721d818" />

2. Hover Effect (Dark Mode)

New Glow Effect:
<img width="1920" height="972" alt="Screenshot (79)" src="https://github.com/user-attachments/assets/6e518477-a2b8-4456-aded-ea012cf6fffa" />

3. Organization Details Page (Dark Mode Fixes)

Step 1: Dark Mode Enabled (Home)
<img width="1920" height="961" alt="Screenshot (80)" src="https://github.com/user-attachments/assets/d51220b5-0d5f-4de4-9415-4801bb8c07ba" />
Step 2: Organization Page in Dark Mode (Now works correctly)
<img width="1920" height="1080" alt="Screenshot (81)" src="https://github.com/user-attachments/assets/ee4a2d84-ba2c-470f-b731-eda5f434c456" />
Step 3: Switching back to Light Mode
<img width="1920" height="975" alt="Screenshot (84)" src="https://github.com/user-attachments/assets/30aadd45-4f2e-4434-ab22-07202a0f6613" />

Step 4: Reload & Verify Light Mode (Retains original look) (After reloading the organization page)
<img width="1920" height="988" alt="Screenshot (86)" src="https://github.com/user-attachments/assets/7c1aa6c8-a3c7-426a-99dd-f73f416b6d6c" />

trivedikavya committed:
in Dark mode Contributor Name Issue Solved

trivedikavya committed:
Added Box shadow In organisation Section In Dark mode as White Mode Have

trivedikavya commented:
One More Thing i added to **enhance UI in Dark mode** .
As **Light mode Have Box-shadow In Organisation** Section around cards , so i added **box-shadow around card In dark mode** to enhance the UI in Dark mode.

Before 
<img width="1920" height="971" alt="Screenshot (90)" src="https://github.com/user-attachments/assets/b294cc6f-29aa-4bfe-b20a-8e164a1c1683" />
After
<img width="1920" height="960" alt="Screenshot (91)" src="https://github.com/user-attachments/assets/569c773d-1fd2-4029-8b2d-0df1b957b105" />

trivedikavya committed:
Merge branch 'master' into Dark-mode

trivedikavya commented:
@nishantwrp Sir any feedback
